src/tlshd/handshake.c

```diff
@@ -43,10 +43,31 @@
 #include <gnutls/abstract.h>
 
 #include <glib.h>
+#include <linux/tls.h>
 
 #include "tlshd.h"
 #include "netlink.h"
 
+#ifdef HAVE_GNUTLS_RECORD_GET_MAX_SEND_SIZE
+int tlshd_set_record_size(gnutls_session_t session)
+{
+	uint16_t max_send_size;
+	int ret;
+
+	max_send_size = gnutls_record_get_max_send_size(session);
+	/* For TLS 1.3 kernel expects us to account for the ContentType */
+	if (gnutls_protocol_get_version(session) == GNUTLS_TLS1_3)
+		max_send_size -= 1;
+
+	ret = setsockopt(gnutls_transport_get_int(session), SOL_TLS,
+			 TLS_TX_MAX_PAYLOAD_LEN, &max_send_size, sizeof(max_send_size));
```

Comment:
This will fail to build on older kernel headers as `TLS_TX_MAX_PAYLOAD_LEN` isn't defined.

You probably want to wrap the function in a kernel version check `ifdef` as well

Reply:
Yeah that's what I mentioned in the comment above as well, do we go the `ifdef` route or do we handle `ENOPROTOOPT` from setsockopt() failing on older kernel with no support.